### PR TITLE
GP2 import scripts

### PR DESCRIPTION
**Pull request by markcsims**
- allows for updates to user members
- refactors update project members
- fixes broken tests
- adds update members to working groups

### CODE REVIEW

**Review by dianaguimaraes — APPROVED** [review]

File: apps/gp2-server/src/controllers/project.controller.ts [hunk 1]
```diff
@@ -47,7 +47,9 @@ export default class Projects implements ProjectController {
     update: gp2.ProjectUpdateRequest,
     loggedInUserId: string,
   ): Promise<gp2.ProjectResponse> {
-    await this.projectDataProvider.update(id, update);
+    await this.projectDataProvider.update(id, {
+      ...update,
```
Comment (jhonny-m) [inline apps/gp2-server/src/controllers/project.controller.ts:51]: whats the difference?

**Review by jhonny-m — APPROVED** [review]

File: apps/gp2-server/src/data-providers/contentful/utils/common.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+) =>
+  Promise.all(
+    idsToDelete.map(async (id) => {
+      const deletable = await environment.getEntry(id);
+      await deletable.unpublish();
+      return deletable.delete();
+    }),
+  );
+export const getIdsToDelete = (
+  previous: { id?: string }[] | undefined,
```
Comment (peterstarling) [inline apps/gp2-server/src/data-providers/contentful/utils/common.ts:15]:
```suggestion
  previous: string[]
```
?

File: apps/gp2-server/src/data-providers/contentful/utils/members.ts [hunk 3]
```diff
@@ -1,41 +1,130 @@
...
       if (!user?.onboarded) {
         return membersList;
       }
-      const groupMember = parseMember<T>(user, member.role);
+      const groupMember = parseMember<T>(member.sys.id, user, member.role);
       return [...membersList, groupMember];
     }, []) || [];
+
+const addNextMember = async <T extends string>(
+  environment: Environment,
```
Comment (peterstarling) [inline apps/gp2-server/src/data-providers/contentful/utils/members.ts:51]: looks to me like all these could be members of a class which takes environment as a dependency, or they could be part of the data provider which has that environment in the scope

File: packages/server-common/src/utils/csv-parse.ts [hunk 4]
```diff
@@ -14,7 +14,7 @@ export const parse =
       pump(
         createReadStream(src),
         csvParse({ from_line: 2 }),
-        through.obj({ maxConcurrency: 10 }, async (chunk, _, callback) => {
+        through.obj({ maxConcurrency: 1 }, async (chunk, _, callback) => {
```
Comment (peterstarling) [inline packages/server-common/src/utils/csv-parse.ts:17]: is this used anywhere else? are we ok to reduce it to 1 like this?

**Review by peterstarling — APPROVED** [review]

**Pull request merged**